sound/soc/sof/intel/hda-dsp.c

```diff
@@ -364,19 +364,12 @@ static int hda_dsp_wait_d0i3c_done(struct snd_sof_dev *sdev)
 
 static int hda_dsp_send_pm_gate_ipc(struct snd_sof_dev *sdev, u32 flags)
 {
-	struct sof_ipc_pm_gate pm_gate;
-	struct sof_ipc_reply reply;
-
-	memset(&pm_gate, 0, sizeof(pm_gate));
+	const struct sof_ipc_pm_ops *pm_ops = sdev->ipc->ops->pm;
 
-	/* configure pm_gate ipc message */
-	pm_gate.hdr.size = sizeof(pm_gate);
-	pm_gate.hdr.cmd = SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_GATE;
-	pm_gate.flags = flags;
+	if (pm_ops && pm_ops->set_pm_state)
+		return pm_ops->set_pm_state(sdev, flags);
 
-	/* send pm_gate ipc to dsp */
-	return sof_ipc_tx_message_no_pm(sdev->ipc, &pm_gate, sizeof(pm_gate),
-					&reply, sizeof(reply));
+	return 0;
```

Comment:
@RanderWang, this patch is a regression: you can only do the switch to the op when that op has been already implemented. 
Before this patch the SOF_IPC_PM_GATE message was sent, with this patch it is not.

Reply:
yes, thanks